### PR TITLE
GH action for iam-client-lib

### PR DESCRIPTION
**Pull request by royki**
- Migrate Travis CI to GitHub Action

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
+name: Deploy
+
+on:
+  push:
+    branches: [ 'develop', 'master' ]
+  pull_request:
+    branches: [ 'develop' ]
```
Comment (jrhender) [inline .github/workflows/deploy.yml:7]: Why not also `master` here?
```suggestion
    branches: [ 'develop' , 'master' ]
```

**Review by jrhender — COMMENTED** [review]
Just one comment. Also need to remove the .travis.yml file please.

As a stretch goal, would be interested if this issue can be fixed while we are working on this 😄  Any thoughts on this @royki ? If it's too much work and out of scope for this, that's okay too https://github.com/energywebfoundation/iam-client-lib/issues/191

Also, would be good to get @JGiter 's review on this PR as well as he is responsible for iam-client-lib now and has some experience with CI.

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+          git fetch origin master:master
+          git fetch origin develop:develop
+
+      - name: Set npm credentials
+        run: echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_TOKEN }}" >> $HOME/.npmrc
+
+      - name: Install package
+        run: |
+          npm ci
+          npm run build --prod
```
Comment (JGiter) [inline .github/workflows/deploy.yml:43]: I guess this is not needed

Comment (JGiter) [inline .github/workflows/deploy.yml:43]:
```suggestion
```

**Review by JGiter — CHANGES_REQUESTED** [review]

**Review by jrhender — APPROVED** [review]
Looks good to me 👍

**Review by JGiter — APPROVED** [review]

**Pull request merged**